File: lab1/java/threaddemo/src/com/company/MainThread.java
package com.company;
import java.math.BigInteger;

public class MainThread extends Thread {
    private final int id;
    private final boolean[] flags;
    private final int index;

    public MainThread(int id, boolean[] flags, int index) {
        this.id = id;
        this.flags = flags;
        this.index = index;
    }

    @Override
    public void run() {
        BigInteger sum = BigInteger.ZERO;
        BigInteger current = BigInteger.ZERO;
        BigInteger step = BigInteger.valueOf(id);
        long count = 0;
        long startTime = System.currentTimeMillis();

        while (!flags[index]) {
            sum = sum.add(current);
            current = current.add(step);
            count++;
        }

        long elapsed = System.currentTimeMillis() - startTime;
        System.out.println("Thread " + id + " | Sum: " + sum + " | Count: " + count + " | Time: " + elapsed + "ms");
    }
}

File: lab1/cs/threaddemosharp/threaddemo/MainThread.cs
using System;
using System.Threading;

namespace threaddemo
{
    public class MainThread
    {
        private readonly int id;
        private readonly bool[] flags;
        private readonly int index;

        public MainThread(int id, bool[] flags, int index)
        {
            this.id = id;
            this.flags = flags;
            this.index = index;
        }

        public void Run()
        {
            long sum = 0, current = 0, step = id, count = 0;
            DateTime startTime = DateTime.Now;

            while (!flags[index])
            {
                sum += current;
                current += step;
                count++;
            }

            double elapsed = (DateTime.Now - startTime).TotalMilliseconds;
            Console.WriteLine($"Thread {id} | Sum: {sum} | Count: {count} | Time: {elapsed:F0}ms");
        }
    }
}
